rules_java_gapic/java_gapic.bzl

```diff
@@ -127,12 +127,14 @@ def java_gapic_library(
         service_yaml = None,
         deps = [],
         test_deps = [],
+        transport = None,
+        java_generator_name = "java_gapic",
```

Comment:
To reuse the same rule for dumping and running from a file (i.e. substitute Main with MainDump).

Reply:
SG, could we please add some documentation here? And since this is intended only for our use, could we please make this fail if the string does not match the generator or the dumpers' values?

Reply:
Renamed the parameter to _java_generator_name (starting with udnerscore), to indicate that it is a private thing. Can we please keep it flexible about which values can be passed (I rather consider it to be a feature).

Reply:
Sure, although some documentation would still be ideal.

Could you please check if the commits have been pushed? I'm still seeing the param without the leading underscore.